src/test/java/de/tum/in/www1/artemis/LearningGoalIntegrationTest.java

```diff
@@ -395,6 +407,22 @@ void deleteLearningGoal_asInstructor_shouldDeleteLearningGoal() throws Exception
         request.get("/api/courses/" + idOfCourse + "/goals/" + idOfLearningGoal, HttpStatus.NOT_FOUND, LearningGoal.class);
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void deleteLearningGoal_witRelatedGoals_shouldReturnBadRequest() throws Exception {
```

Comment:
Optional request: Maybe add a comment, why this test case returns a bad request (what condition was not fulfilled)